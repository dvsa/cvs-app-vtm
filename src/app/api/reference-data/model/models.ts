export * from './reference-data.model';
