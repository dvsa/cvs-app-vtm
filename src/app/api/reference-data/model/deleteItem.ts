/**
 * Reference Data Microservice
 * This is the API spec for the reference data service. The reference data will be stored in a AWS DynamoDB database.
 *
 * OpenAPI spec version: 1.0.0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 */

export interface DeleteItem { 
    success?: boolean;
}