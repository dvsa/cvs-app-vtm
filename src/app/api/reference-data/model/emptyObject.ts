/**
 * Reference Data Microservice
 * This is the API spec for the reference data service. The reference data will be stored in a AWS DynamoDB database.
 *
 * OpenAPI spec version: 1.0.0
 * Contact: test@test.com
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 */

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface EmptyObject {
}
