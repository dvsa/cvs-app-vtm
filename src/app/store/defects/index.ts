export * from './actions/defects.actions';
export * from './reducers/defects.reducer';
export * from './selectors/defects.selectors';
