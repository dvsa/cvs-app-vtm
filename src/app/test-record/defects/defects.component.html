<dl *ngFor="let defect of testType.defects" class="govuk-summary-list">
  <h2 class="govuk-heading-l">
    <span class="govuk-!-margin-top-3" id="test-deficiencyRef"> {{ defect.deficiencyRef }}</span>
    -
    <span
      class="govuk-tag govuk-!-font-size-24"
      id="test-deficiencyCategory"
      [class.tag--black]="defect.deficiencyCategory === 'dangerous'"
      [class.tag--red]="defect.deficiencyCategory === 'major'"
    >
      {{ defect.deficiencyCategory }}
    </span>
  </h2>
  <div class="govuk-summary-list__row">
    <dt class="govuk-summary-list__key">
      Category
    </dt>
    <dd class="govuk-summary-list__value" id="test-imNumber-imDescription">
      {{
        (defect.imNumber ? defect.imNumber : '') +
          ' ' +
          (defect.imDescription ? defect.imDescription : '') | DefaultNullOrEmpty
      }}
    </dd>
  </div>
  <div class="govuk-summary-list__row">
    <dt class="govuk-summary-list__key">
      Item
    </dt>
    <dd class="govuk-summary-list__value" id="test-itemNumber-itemDescription">
      {{
        (defect.itemNumber ? defect.itemNumber : '') +
          ' ' +
          (defect.itemDescription ? defect.itemDescription : '') | DefaultNullOrEmpty
      }}
    </dd>
  </div>
  <div class="govuk-summary-list__row">
    <dt class="govuk-summary-list__key">
      Description
    </dt>
    <dd class="govuk-summary-list__value" id="test-deficencyId">
      {{
        (defect.deficiencyId ? defect.deficiencyId : '') +
          ' ' +
          (defect.deficiencySubId ? defect.deficiencySubId : '') +
          ' ' +
          (defect.deficiencyText ? defect.deficiencyText : '') | DefaultNullOrEmpty
      }}
    </dd>
  </div>
  <div class="govuk-summary-list__row">
    <dt class="govuk-summary-list__key">
      Location
    </dt>
    <dd class="govuk-summary-list__value" id="test-location">
      {{
        (defect.additionalInformation.location.vertical
          ? (defect.additionalInformation.location.vertical | titlecase)
          : '') +
          (defect.additionalInformation.location.horizontal
            ? '/' + (defect.additionalInformation.location.horizontal | titlecase)
            : '') +
          (defect.additionalInformation.location.lateral
            ? '/' + (defect.additionalInformation.location.lateral | titlecase)
            : '') +
          (defect.additionalInformation.location.longitudinal
            ? '/' + (defect.additionalInformation.location.longitudinal | titlecase)
            : '') +
          (defect.additionalInformation.location.rowNumber
            ? '/' + (defect.additionalInformation.location.rowNumber.toString() | titlecase)
            : '') +
          (defect.additionalInformation.location.seatNumber
            ? '/' + (defect.additionalInformation.location.seatNumber.toString() | titlecase)
            : '') +
          (defect.additionalInformation.location.axleNumber
            ? '/' + (defect.additionalInformation.location.axleNumber | titlecase)
            : '') | DefaultNullOrEmpty
      }}
    </dd>
  </div>
  <div class="govuk-summary-list__row">
    <dt class="govuk-summary-list__key">
      Repaired during test
    </dt>
    <dd class="govuk-summary-list__value" id="test-defect-prs">
      {{ defect.prs | DefaultNullOrEmpty | titlecase }}
    </dd>
  </div>
  <div class="govuk-summary-list__row">
    <dt class="govuk-summary-list__key">
      Prohibition issued
    </dt>
    <dd class="govuk-summary-list__value" id="test-prohibition">
      {{ defect.prohibitionIssued | DefaultNullOrEmpty | titlecase }}
    </dd>
  </div>
  <div class="govuk-summary-list__row">
    <dt class="govuk-summary-list__key">
      Additional defect details
    </dt>
    <dd class="govuk-summary-list__value" id="test-additionalInfo">
      {{ defect.additionalInformation.notes | DefaultNullOrEmpty | titlecase }}
    </dd>
  </div>
</dl>
