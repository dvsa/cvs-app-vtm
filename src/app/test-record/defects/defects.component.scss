
.govuk-summary-list {

  .tag-custom {
    padding: 1px 10px 1px 10px;
    vertical-align: middle;
  }

  & .tag--black {
    background-color: #0b0c0c !important;
    @extend .tag-custom;
  }

  & .tag--red {
    background-color: #d4351c !important;
    @extend .tag-custom;
  }

  & .govuk-summary-list__row:last-child {
    & .govuk-summary-list__value, .govuk-summary-list__key {
      border-bottom: none !important;
    }
  }
}

