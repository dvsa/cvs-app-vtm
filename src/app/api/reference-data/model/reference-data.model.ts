import { ReferenceDataResourceType } from '@models/reference-data.model';

export type ReferenceDataItem = {
  resourceType: ReferenceDataResourceType;
  resourceKey: string | number;
};

export type ReferenceDataList = Array<ReferenceDataItem>;

export type ReferenceDataApiResponse = {
  data: ReferenceDataList;
};
