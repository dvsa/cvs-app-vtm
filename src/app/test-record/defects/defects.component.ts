import { Component, OnInit, ChangeDetectionStrategy, Input } from '@angular/core';
import { TestType } from '@app/models/test.type';
import { TestResultModel } from '@app/models/test-result.model';

@Component({
  selector: 'vtm-defects',
  templateUrl: './defects.component.html',
  styleUrls: ['./defects.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class DefectsComponent implements OnInit {
  @Input() testType: TestType;
  @Input() testRecord: TestResultModel;

  constructor() {}

  ngOnInit() {}

  concatStrings(strings, concatBy): string {
    const stringsToJoin = strings.length > 1 ? strings.filter(item => item) : [''];
    return stringsToJoin.join(concatBy);
  }
}
