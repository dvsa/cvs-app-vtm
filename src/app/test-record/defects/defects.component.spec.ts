import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { DefectsComponent } from './defects.component';
import {SharedModule} from '@app/shared/shared.module';
import {TechRecordHelpersService} from '@app/technical-record/tech-record-helpers.service';
import {TestType} from '@app/models/test.type';
import {TestResultModel} from '@app/models/test-result.model';

describe('DefectsComponent', () => {
  let component: DefectsComponent;
  let fixture: ComponentFixture<DefectsComponent>;
  const testType = {} as TestType;
  const testRecord = {} as TestResultModel;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [ SharedModule ],
      declarations: [ DefectsComponent ],
      providers: [
        TechRecordHelpersService
      ]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(DefectsComponent);
    component = fixture.componentInstance;
    component.testType = testType;
    component.testRecord = testRecord;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
});
