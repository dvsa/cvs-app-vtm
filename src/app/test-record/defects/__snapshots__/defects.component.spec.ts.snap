// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefectsComponent should create 1`] = `
<vtm-defects
  testRecord={[Function Object]}
  testType={[Function Object]}
>
  
</vtm-defects>
`;
